python/lsst/daf/butler/butler.py

```diff
@@ -140,6 +146,13 @@ def makeRepo(root, config=None, standalone=False, createRegistry=True):
         os.error
             Raised if the directory does not exist, exists but is not a
             directory, or cannot be created.
+
+        Notes
+        -----
+        Note that when ``standalone=False`` (the default), the configuration
+        search path (see `ConfigSubset.defaultSearchPaths`) that was used to
+        construct the repository should also be used to construct any Butlers
+        to it to avoid configuration inconsistencies.
```

Comment:
I think this is a very dangerous way to handle things and should be re thought in the future